app/main/views/auth.py

```diff
@@ -82,11 +131,15 @@ def request_email_address():
             if idp is None:
                 return redirect_to_broker(IDP_OF_LAST_RESORT)
 
-            if isinstance(idp, list):
-                session['idp_choices'] = idp
+            session['idp_choices'] = [item['id'] for item in idp]
+
+            if len(idp) > 1:
                 return redirect(url_for('.select_idp'))
 
-            return redirect_to_broker(idp)
+            session['resolved_idp'] = idp[0].get("id")
+            session['department_name'] = idp[0].get("name")
+
+            return redirect(url_for('.confirm_dept'))
 
         else:
```

Comment:
Sometimes this will be false, or 'no' in this case so the redirection will be to select_dept route